src/main/java/com/g4s8/mime/test/HmMimeHasParam.java

```diff
@@ -0,0 +1,109 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.g4s8.mime.test;
+
+import com.g4s8.mime.MimeType;
+import java.io.IOException;
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+/**
+ * MIME param matcher.
+ *
+ * @author Viciouss (xviciouss@gmail.com)
+ * @version $Id$
+ * @since 1.0
+ */
+public final class HmMimeHasParam extends TypeSafeMatcher<MimeType> {
+
+    /**
+     * The parameter name to check.
+     */
+    private final String name;
+
+    /**
+     * Expected matcher.
+     */
+    private final Matcher<String> expected;
+
+    /**
+     * Ctor.
+     * @param name The name of the parameter to be matched.
+     * @param expected The actual value that should be in this parameter.
+     */
+    public HmMimeHasParam(final String name, final String expected) {
+        super(MimeType.class);
+        this.name = name;
+        this.expected = CoreMatchers.equalTo(expected);
+    }
+
+    /**
+     * Ctor.
+     * @param name The name of the parameter to be matched.
+     * @param expected A matcher that can be used to check the param.
+     */
+    public HmMimeHasParam(final String name, final Matcher<String> expected) {
+        super(MimeType.class);
+        this.name = name;
+        this.expected = expected;
+    }
+
+    @Override
+    public boolean matchesSafely(final MimeType type) {
+        try {
+            final String actual = type.params().get(this.name);
+            return this.expected.matches(actual);
+        } catch (final IOException err) {
+            throw new UnsupportedOperationException(
+                "Matcher failed due to error",
+                err
+            );
+        }
+    }
+
+    @Override
+    public void describeTo(final Description description) {
+        description.appendText("MIME with param ")
+            .appendValue(this.name)
+            .appendText(" = ");
+        this.expected.describeTo(description);
+    }
+
+    @Override
+    public void describeMismatchSafely(
+        final MimeType item,
+        final Description desc
+    ) {
+        try {
+            desc.appendText("was param ")
+                .appendValue(this.name)
+                .appendText(" = ")
```

Comment:
Parameter description here is same as in `describeTo` method: `key = param`, so lets move key-value formatting into `SelfDescribing` class (I think it should be nested private nested class) which will contain param's name and `SelfDescribing` value (because it may be matcher in `describeTo` or `SelfDescribingValue<>` in `describeMismatchSafely`) and put it into description as argument:
```java
description.appendText("MIME with param ")
    .appendDescriptionOf(new ParamDescription(this.name, this.expected)) // for describeTo

desc.appendText("was param")
    .appendDescriptionOf(
        new ParamDescription(
            this.name, 
            new SelfDescribingValue<>(item.params().get(this.name))
        )
    ) // for describeMismatchSafely
```